File: Server/aoRssFeed2/aoRssFeed2/Views/RssFeedLinkListClass.vb

Option Strict On
Option Explicit On

Imports System.Linq
Imports Contensive.Addons.aoRssFeed2.Controllers
Imports Contensive.BaseClasses

Namespace Views
    '
    Public Class RssFeedLinkListClass
        Inherits AddonBaseClass
        '
        '=====================================================================================
        ''' <summary>
        ''' AddonDescription
        ''' </summary>
        ''' <param name="CP"></param>
        ''' <returns></returns>
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim rssfeedList As List(Of Models.Db.RSSFeedModel) = Models.Db.RSSFeedModel.createList(CP, "", "id desc")
                If (rssfeedList.Count = 0) Then
                    result = CP.Html.p("There are currently no public RSS Feeds available.")
                Else
                    Dim liList As String = ""
                    For Each rssfeed As Models.Db.RSSFeedModel In rssfeedList
                        liList += CP.Html.li(genericController.getFeedLink(CP, rssfeed), "", "RSSFeedListItem")
                    Next
                    result = CP.Html.ul(liList, "", "RSSFeedList")
                    result = CP.Html.div(result, "", "RSSFeedWrapper")
                End If
            Catch ex As Exception
                CP.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace

File: Server/aoRss/Views/RssFeedLinkListClass.cs
using System;
using Contensive.Addons.Rss.Controllers;
using Contensive.BaseClasses;
using Contensive.Models.Db;

namespace Contensive.Addons.Rss.Views {
    // 
    public class RssFeedLinkListClass : AddonBaseClass {
        // 
        // =====================================================================================
        /// <summary>
        /// AddonDescription
        /// </summary>
        /// <param name="CP"></param>
        /// <returns></returns>
        public override object Execute(CPBaseClass CP) {
            string result = "";
            try {
                var rssfeedList = DbBaseModel.createList<Models.Db.RSSFeedModel>(CP, "", "id desc");
                if (rssfeedList.Count == 0) {
                    result = CP.Html.p("There are currently no public RSS Feeds available.");
                } else {
                    string liList = "";
                    foreach (Models.Db.RSSFeedModel rssfeed in rssfeedList)
                        liList += CP.Html.li(GenericController.getFeedLink(CP, rssfeed), "", "RSSFeedListItem");
                    result = CP.Html.ul(liList, "", "RSSFeedList");
                    result = CP.Html.div(result, "", "RSSFeedWrapper");
                }
            } catch (Exception ex) {
                CP.Site.ErrorReport(ex);
            }
            return result;
        }
    }
}